src/lib/mina_base/account.ml

```diff
@@ -696,6 +696,37 @@ let create_time_locked public_key balance ~initial_minimum_balance ~cliff_time =
     ~vesting_period:Global_slot.(succ zero)
     ~vesting_increment:initial_minimum_balance
 
+let initial_minimum_balance Poly.{ timing; _ } =
+  match timing with
+  | Timing.Untimed ->
```

Comment:
Shouldn't these return an `option` type? As written, they're encoding `None` as `zero`, essentially.

Allowing the `vesting_period` to be `zero` breaks an invariant (see `Account.create_timed`).

Reply:
My thinking was that an untimed account is just a special case of a timed account where initial minimum balance is `0` and hance other properties don't matter, because there's nothing to vest. But I agree it might be confusing, so I changed the types of these functions to indeed return options. I also added helpers next to the tests to unpack these options for convenience. See 03d8ba9dee.